plugin-markerview/src/main/java/com/mapbox/mapboxsdk/plugins/markerview/MarkerViewManager.java

```diff
@@ -0,0 +1,67 @@
+package com.mapbox.mapboxsdk.plugins.markerview;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.UiThread;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import timber.log.Timber;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MarkerViewManager implements MapView.OnMapChangedListener {
+
+  private final MapView mapView;
+  private final MapboxMap mapboxMap;
+  private final List<MarkerView> markers = new ArrayList<>();
+  private boolean initalised;
+
+  public MarkerViewManager(MapView mapView, MapboxMap mapboxMap) {
+    this.mapView = mapView;
+    this.mapboxMap = mapboxMap;
+  }
+
+  public void onDestroy() {
```

Comment:
This call is missing in the example. This method should probably be limited to the main thread as well.